modules/programs/terminator.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.terminator;
+
+  toValue = val:
+    if val == null then
+      "None"
+    else if val == true then
+      "True"
+    else if val == false then
+      "False"
+    else
+      ''"${toString val}"'';
+
+  toConfigObject = let
+    toKey = depth: key:
+      if depth == 0 then key else toKey (depth - 1) "[${key}]";
+    go = depth: obj:
+      flatten (mapAttrsToList (key: val:
+        if isAttrs val then
+          [ (toKey depth key) ] ++ go (depth + 1) val
+        else
+          [ "${key} = ${toValue val}" ]) obj);
+  in obj: concatStringsSep "\n" (go 1 obj);
```

Comment:
Can we use one of the existing config formats? This looks like it's very similar to an INI file, and nixpkgs has a generator for that.

Reply:
The `toINI` generator only supports a single level. According to the [manpage](https://linux.die.net/man/5/terminator_config) the format is called `ConfigObject` which seems to be some python specific format.